File: 20180712/Min Cost Climbing Stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        // Python version 100% beats??
        // if (cost.length < 3){
        //     Arrays.sort(cost);
        //     return cost[0];
        // }
        //  List<Integer> res = new ArrayList<>();
        // for (int i : cost) {
        //     res.add(i);
        // }
        // res.add(0);
        // res.toArray();
        // for(int i = 2; i < res.size(); i ++){
        //     int value = res.get(i);
        //     res.set(i, value + Math.min(res.get(i - 1), res.get(i - 2)));
        // }
        // return res.get(res.size() - 1);
        if(cost.length < 3){
            Arrays.sort(cost);
            return cost[0];
        }
        int[] res = new int[cost.length + 1];
        res[0] = 0;
        res[1] = 0;
        for (int i = 2; i < res.length; i ++){
            res[i] = Math.min(res[i - 1] + cost[i - 1], res[i - 2] + cost [i - 2]);
        }
        return res[res.length - 1];
    }
}

File: 20180712/Min Cost Climbing Stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        # dp = [0] * len(cost)
        # for i in range(2,len(dp),1):
        #     dp[i] = min(dp[i-1] + cost[i-1], dp[i-2] + cost[i-2])
        # return min(dp[-1] + cost[-1], dp[-2] + cost[-2])
        if len(cost) < 3:
            return min(cost)
        cost.append(0)
        for i in range(2, len(cost), 1):
            cost[i] += min(cost[i-1], cost[i-2])
        return cost[-1]        